src/global/pid_lut/pid_lut.cc

```diff
@@ -210,6 +177,40 @@ void InitPlugin(JApplication* app) {
       },
       dirc_pid_cfg, app));
 
+  // Inject particles from other sources without PID detectors so they are contained
+  // as a particle in the ReconstructedChargedParticle collection rather than needing
+  // a subset collection. This should be fixed in the future.
+
+  app->Add(
+      new JOmniFactoryGeneratorT<CollectionCollector_factory<edm4eic::ReconstructedParticle, true>>(
+          "ReconstructedWithPFRICHTOFDIRCLOWQ2PIDChargedParticles",
+          {"ReconstructedChargedWithPFRICHTOFDIRCPIDParticles",
+           "TaggerTrackerReconstructedParticles"},
+          {"ReconstructedWithPFRICHTOFDIRCLOWQ2PIDChargedParticles"}, app));
+
+  app->Add(new JOmniFactoryGeneratorT<
+           CollectionCollector_factory<edm4eic::MCRecoParticleAssociation, true>>(
+      "ReconstructedChargedWithPFRICHTOFDIRCLOWQ2PIDParticleAssociations",
+      {"ReconstructedChargedWithPFRICHTOFDIRCPIDParticleAssociations",
+       "TaggerTrackerReconstructedParticleAssociations"},
+      {"ReconstructedChargedWithPFRICHTOFDIRCLOWQ2PIDParticleAssociations"}, app));
```

Comment:
I think, this is broken. See https://eicrecon.epic-eic.org/pr/2018/capybara/rec_dis_18x275_minQ2=0_craterlake_18x275/index.html#_ReconstructedParticles_particleIDs
<img width="797" height="300" alt="image" src="https://github.com/user-attachments/assets/6f745262-5702-4a09-9244-6bd5199a2fd5" />
This loses 2 out of 4 PID LUTs.

Looking again at logic in
https://github.com/eic/EICrecon/blob/6e3493bd5a2b1d851660e66a72561af2109d263f/src/algorithms/pid_lut/PIDLookup.cc#L67-L72
I don't see what is wrong yet. It should match entries just fine, if it's just a subset collection, and it should be:
https://github.com/eic/EICrecon/blob/6e3493bd5a2b1d851660e66a72561af2109d263f/src/algorithms/meta/CollectionCollector.h#L39

Reply:
Further comments, I'm not sure what changed to cause the changes between your original figure and the one which is in the link now. 

Of the originally missing ParticleIDs. One of them was the `FarBackwardPhaseSpacePIDParticleIDs` which has been removed and the other the `RICHEndcapNParticleIDs`.

For some reason the `RICHEndcapNParticleIDs` appeared to be adding a PID to the LowQ2 particles. The `pfrich.lut` file covers values all the way up to pi based however the binning provided in the pid_lut plugin doesn't appear to. Looking at the implementation the binning in the config should provide the histogram limits and the file should just fill it.

Reply:
Updated image after #2083 from: 
 https://eicrecon.epic-eic.org/pr/2018/capybara/rec_dis_18x275_minQ2=0_craterlake_18x275/index.html#_ReconstructedParticles_particleIDs
<img width="904" height="354" alt="image" src="https://github.com/user-attachments/assets/0fb6563f-b2e3-4f7b-974e-f9dc173b7efb" />

Shows only the `FarBackwardPhaseSpacePIDParticleIDs` collection has been removed an no changes to the content of the others. I can't comment on why the previous plots appeared so different with totally different counts in even the completely filled ParticleIDs.